### DISCUSSION TITLE
[Feature] Considering to have customizing product's auto-filled fields in modules advanced section

### DISCUSSION THREAD

**Discussion by kseibi** [YetiForceCompany/YetiForceCRM#14191] [Feature improvement]
## 🚀 feature request

It would be very, very great to have auto-filled fields from products into the modules advanced sections, the current system only auto-fills the price, the cost, and some other margin/discount/tax information, we would be extremely grateful if we could design more customized pre-filled data like required delivery time, stock availability (Yes or No), gross weight, package quantity, Package dimension, Product Material, and so on to be added into the advanced module section, in that case, we can export more types of PDF documents right from the CRM, not only invoices, we can have packing lists, production data requirements, customs-clearance form, shipping form, stocklist, etc... without having to use excel anymore.

It would also be greater to have an option to add a 50x50px image 

### 🚀 Is your feature request related to a problem? Please describe.

At the moment we can only make invoices from the system, we already have installed these modules for other documents, but we cannot generate them from the system because we can only make it in excel and upload it as a document.

**Comment by KatarzynaUlichnowska** [comment]
Hello, each request for a new functionality has to be submitted separately.

**Comment by vovpff** [comment]
Everything what you requested available in system.

**Comment by kseibi** [comment]
In the current system, we can manually add any fields to the sales order (and other) advanced modules section by adding blank fields, and then have to manually fill it one by one in the sales order module, **it's very great**, but there is no possibility to have auto-filled data from the Product Module directly when we add the item, for example, the below photo, we have added comments and total cartons fields, it would be very great if we could pre-fill the comment and total cartons in the product module, and get them automatically written on the below image, and we can edit them if necessary right from the sales-order module.

![Click](https://user-images.githubusercontent.com/74230891/101315903-6ecb5000-3896-11eb-80e7-9fc2d5dee9fb.jpg)

If the function is already available in the system, great, please let me know how it is made.

**Comment by vovpff** [comment]
@kseibi you need to add one more string to `u_yf_ssingleorders_invmap` database table to map Total cartons row of Sales Order to Total cartons field of product.
This is looks not easy but this works.

Some features available in system but should be added manualy and not available from admin back end.

**Comment by kseibi** [comment]
thank you very much, It worked really great, but the mapped field is closed and uneditable, is there a hint which file should I look up to make it editable? thank you again for your support.

![Click8](https://user-images.githubusercontent.com/74230891/101585623-1a9da880-3a1b-11eb-82e3-a6a036d1df10.jpg)

**Comment by vovpff** [comment]
Looks like you set box size field as read only
